src/figuresClasses.ts

```diff
@@ -1,11 +1,73 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a = 0,
+    public b = 0,
+    public c = 0,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0 || a >= b + c || b >= a + c || c >= a + b) {
+      throw new Error(`sides 1, 2 and 3 can't form a triangle`);
+    }
+  }
+
+  getArea(): number {
+    const s = (1 / 2) * (this.a + this.b + this.c);
+    const area = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+    return Math.floor(area * 100) / 100;
+  }
+}
+
+export class Circle implements Figure {
+  public shape: Shape = 'circle';
 
-export class Circle implements Figure {}
+  constructor(
+    public color: Color,
+    public radius: number = 0,
+  ) {
+    if (radius <= 0) {
+      throw new Error(`it can't be a circle`);
```

Comment:
[CODE STYLE] - If you throw an error try to provide its detailed description in the error message. The message `it can't be a circle` could be improved by specifying that the radius must be greater than zero.